### DISCUSSION TITLE
Issue with the translation symmetries

### DISCUSSION THREAD

**Discussion by kaushman1996** [QuSpin/QuSpin#669] [General]
Dear Quspin community,

I tried implementing a simple t-U-V model over a triangular lattice. I implemented translation symmetries on top of it, but I  am getting weird results far from the results with full basis (this result I understand and is correct too). I suspect there is something wrong with the symmetries for spinful fermion basis. I have earlier tried the same lattice with spin model too, and there also I faced issues with translation symmetries. I have attached the lattice image and the minimal code snippet. Please help me or suggest me something with this.   

Regards
Aman
![photo_2024-03-25 18 33 58](https://github.com/QuSpin/QuSpin/assets/49250603/78fe4664-3298-41c4-9407-84d1f5d98dd5)

`from quspin.basis import spinful_fermion_basis_general
from quspin.operators import hamiltonian
import numpy as np
###### define model parameters ######
N= 12# number of sites for spin 1
#
J=1.0 # hopping matrix element
U=100.0 # onsite interaction
V=30.0
#
###### setting up user-defined BASIC symmetry transformations for 2d lattice ######
T_x=[1,2,7,4,5,6,10,8,9,0,11,3]
T_y=[4,5,6,7,8,9,0,10,11,3,1,2]
T_z=[10,11,3,0,1,2,7,4,5,6,8,9]
###### setting up bases ######
basis = spinful_fermion_basis_general(N,Nf=(2,2),kxblock=(T_x,0),kyblock=(T_y,0))
basis1 = spinful_fermion_basis_general(N,Nf=[(2,2)])

# setting up site-coupling lists for simple case
hopping_left =[[-J,i,T_x[i]] for i in range(N)] + [[-J,i,T_y[i]] for i in range(N)] + [[-J,i,T_z[i]] for i in range(N)]
print(hopping_left)
hopping_right=[[+J,i,T_x[i]] for i in range(N)] + [[+J,i,T_y[i]] for i in range(N)] + [[+J,i,T_z[i]] for i in range(N)]
interaction=[[U,i,i] for i in range(N)]
interaction1=[[V,i,T_x[i]] for i in range(N)] + [[V,i,T_y[i]] for i in range(N)] + [[V,i,T_z[i]] for i in range(N)]
interaction2=[[V,T_x[i],i] for i in range(N)] + [[V,T_y[i],i] for i in range(N)] + [[V,T_z[i],i] for i in range(N)]

static=[["+-|",hopping_left], # spin up hops to left
                ["-+|",hopping_right], # spin up hops to right
                ["|+-",hopping_left], # spin down hopes to left
                ["|-+",hopping_right], # spin up hops to right
                ["n|n",interaction], # spin up-spin down interaction
                ["n|n",interaction1], # spin up-spin down interaction
                ["nn|",interaction1], # spin up-spin down interaction
                ["|nn",interaction1], # spin up-spin down interaction
                ["n|n",interaction2]] # spin up-spin down interaction
# simulation with symmetry used basis
H=hamiltonian(static,[],basis=basis,dtype=np.float64)
E_GS,psi_GS = H.eigsh(k=1,which="SA")
for i in range(12):
    zz=[[1,i]]
    static1=[["n|",zz],["|n",zz]]
    Num= hamiltonian(static1,[],basis=basis,dtype=np.float64)
    print('check expectation of Num',i,"**", Num.expt_value(psi_GS))
# simulation with full basis
H=hamiltonian(static,[],basis=basis1,dtype=np.float64)
E_GS,psi_GS = H.eigsh(k=1,which="SA")
for i in range(12):
    zz=[[1,i]]
    static1=[["n|",zz],["|n",zz]]
    Num= hamiltonian(static1,[],basis=basis1,dtype=np.float64)
    print('check expectation of Num',i,"**", Num.expt_value(psi_GS))`

**Comment by mgbukov** [comment]
It looks like you're using an operator -- 'zz' -- which breaks translation
symmetry.

**Reply by kaushman1996** [reply]
Yes, thanks for the response; I realized it after posting the question and got the fix also.
